src/index.spec.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react'
+import { spy } from 'chai'
+import { mount } from 'enzyme'
+import ReactDOMServer from 'react-dom/server'
+
+import IsomorphicLoader from './index'
+
+/* eslint-env mocha */
+/* eslint-disable
+  no-unused-expressions,
+  react/jsx-filename-extension,
+  import/no-extraneous-dependencies,
+  react/prefer-stateless-function,
+  max-len
+*/
+
+class Child extends Component {
+  render() {
+    return (<div />)
+  }
+}
+
+const initSpies = () => {
+  spy.on(IsomorphicLoader.prototype, 'componentDidMount')
+  spy.on(IsomorphicLoader.prototype, 'render')
+  spy.on(Child.prototype, 'render')
+}
+
+const render = (renderer, strict = true, callback) => renderer(
+  <IsomorphicLoader library="react" strict={strict}>
+    {lib => {
+      if (callback && callback === '[object Function]') callback(lib)
+      return <Child />
+    }}
+  </IsomorphicLoader>
+)
+
+describe('IsomorphicLoader', () => {
+  describe('Serverside', () => {
+    it('should call render function once but not componentDidMount function', () => {
+      initSpies()
+
+      render(ReactDOMServer.renderToString)
+
+      IsomorphicLoader.prototype.componentDidMount.should.not.have.been.called.once
```

Comment:
If the componnentDidMount function is called twice this will pass.
You may want to use
```js
IsomorphicLoader.prototype.componentDidMount.should.not.have.been.called()
```